selenium_tests/publish_courses/page_objects/main_page.py

```diff
@@ -0,0 +1,62 @@
+"""
+This page models the main (landing) page of the Publish Courses Tool
+"""
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+from selenium_tests.publish_courses.page_objects.base_page_object \
+    import PublishCoursesBasePageObject
+
+
+class Locators(object):
+    PUBLISH_ALL_BUTTON = (By.ID, "publishAllButton")
+    TERM_DROPDOWN = (By.ID, "dropdownMenuTerm")
+    RESULT_CONFIRMATION = (By.ID, "result-message")
+
+    @classmethod
+    def FIND_ELEMENT_WITH_TEXT(cls, text_value):
+        """Locates the xpath, given the text value"""
+        return By.XPATH, "//a[contains(.,'{}')]".format(text_value)
```

Comment:
This finds anchor elements, no? so maybe `FIND_ELEMENT_WITH_TEXT` could be `FIND_LINK_WITH_TEXT` or something?